### ISSUE TITLE
Command in wav2vec documentation no longer works: --criterion binary_cross_entropy is unrecognized.

### ISSUE THREAD

**Issue by dmjef** [pytorch/fairseq#2434]
## 🐛 Bug

On [this page](https://github.com/pytorch/fairseq/blob/master/examples/wav2vec/README.md), the command to pre-train wav2vec (version 1) fails because `--criterion binary_cross_entropy` is no longer recognized. Changing that argument to `cross_entropy` fails due to a NotImplementedError - it's not obvious what to use.

### To Reproduce

Run the following command, taken verbatim from the wav2vec readme, with the proper code and data setup:
`python train.py /manifest/path --save-dir /model/path --num-workers 6 --fp16 --max-update 400000 --save-interval 1 --no-epoch-checkpoints \
--arch wav2vec --task audio_pretraining --lr 1e-06 --min-lr 1e-09 --optimizer adam --max-lr 0.005 --lr-scheduler cosine \
--conv-feature-layers [(512, 10, 5), (512, 8, 4), (512, 4, 2), (512, 4, 2), (512, 4, 2), (512, 1, 1), (512, 1, 1)] \
--conv-aggregator-layers [(512, 2, 1), (512, 3, 1), (512, 4, 1), (512, 5, 1), (512, 6, 1), (512, 7, 1), (512, 8, 1), (512, 9, 1), (512, 10, 1), (512, 11, 1), (512, 12, 1), (512, 13, 1)] \
--skip-connections-agg --residual-scale 0.5 --log-compression --warmup-updates 500 --warmup-init-lr 1e-07 --criterion binary_cross_entropy --num-negatives 10 \
--max-sample-size 150000 --max-tokens 1500000 --skip-invalid-size-inputs-valid-test`

The command fails with the following error message:
`train.py: error: argument --criterion: invalid choice: 'binary_cross_entropy' (choose from 'legacy_masked_lm_loss', 'masked_lm', 'sentence_prediction', 'label_smoothed_cross_entropy', 'nat_loss', 'composite_loss', 'ctc', 'cross_entropy', 'adaptive_loss', 'sentence_ranking', 'wav2vec', 'label_smoothed_cross_entropy_with_alignment', 'vocab_parallel_cross_entropy')`

### Additional context

This is happening with the current version on master but not version cf87f759b9f09769dc416761738cee72382252df. So this issue was introduced with word2vec 2.0.

**Comment by Squire-tomsk** [comment]
I suppose answer in this commit: https://github.com/pytorch/fairseq/commit/621e834103b13318cb48d41fc713b580f0da6b24#diff-247f5afabd2344233e6500e2b5da64d8

fairseq/criterions/binary_cross_entropy.py → fairseq/criterions/wav2vec_criterion.py

**Comment by myleott** [comment]
Fixed in 983163494663e24b611f1ba8d5d47a3edc00e2e5
